src/functions-and-arrays.js

```diff
@@ -1,41 +1,155 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(arg1, arg2) {
+  if(arg1 > arg2){
+    return arg1;
+  }else if(arg2 > arg1){
+    return arg2;
+  }else{
+    return arg1 && arg2;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(arr) {
+
+  longestWord = "";
+
+  if(!arr.length){
+    return null;
+  }else if(arr.length === 1){
+    return arr[0];
+  }else{
+    for(let i = 0; i < arr.length; i++){
+      if(arr[i].length > longestWord.length){
+        longestWord = arr[i];
+      }
+    }
+    return longestWord;
+  }
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(arrNumbers) {
+  if(!arrNumbers.length){
+    return 0;
+  }if(arrNumbers.length === 1){
+    return arrNumbers[0];
+  }
+
+  let sum = 0;
+  for(let i = 0; i < arrNumbers.length; i++){
+    if(arrNumbers[i] === 0){
+      return 0;
+    }else{
+      sum = sum + arrNumbers[i];
+    }
+  }
+
+  return sum;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(array1) {
+
+  if(array1.length === 0){
+    return 0;
+  }
+
+  let sum = 0;
+
+  for(let i = 0; i < array1.length; i++){
+    if(typeof array1[i] === "number"){
+      sum += array1[i];
+    }else if(typeof array1[i] === "string"){
+      sum += array1[i].length;
+    }else if(array1[i] === true){
+      sum += 1;
+    }else if(array1[i] === false){
+      sum = sum
+    }else {
+      throw new Error("Data type is not valid")
+    }
+  }
+
+  return sum;
+
+
+}
+  
+
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(arrNumbersAverage) {
+
+  let sum = 0;
+  
+    if(!arrNumbersAverage.length){
+      return null;
+    }else{
+      for(let i = 0; i < arrNumbersAverage.length; i++){
+        sum = sum + arrNumbersAverage[i];
+      }
+      return sum/arrNumbersAverage.length;
+    }
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(stringsArr) { 
+
+  let sum = 0;
+
+  if(!stringsArr.length){
+    return null;
+  }else{
+    for(let i = 0; i < stringsArr.length; i++){
+      sum = sum + stringsArr[i].length;
+  }
+
+    return sum/stringsArr.length;
+  }
+}
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(array2) {
+
```

Comment:
As you already did the sum() function before, you could have done this iteration like this:

```
function avg (arr) {
  if (arr.length === 0) return null;
  else return sum(arr) / arr.length;
}
```